### DISCUSSION TITLE
Setup Selenium grid

### DISCUSSION THREAD

**Discussion by MartinVWWA** [SeleniumHQ/docker-selenium#2201] [Q&A] [answered]
Hello, I have another query regarding setting up a classic Selenium Grid. My aim is to configure it in such a way that each session launches only one browser for a single test, closing thereafter. This functionality resembles that of the dynamic Selenium Grid. Could you please advise on what parameters I should set in the YAML configuration file to achieve this? Thank you.

I have two machines at my disposal. Currently, I'm encountering a scenario where two browsers initiate within a single session. It's imperative to prevent the second session from launching if the first browser has completed its task.

<img width="399" alt="mRemoteNG_dZekmym4MN" src="https://github.com/SeleniumHQ/docker-selenium/assets/166417911/d323cdb0-17c2-4018-8575-b7129fd00b1c">

<img width="332" alt="ApplicationFrameHost_Z3DWl8h0SL" src="https://github.com/SeleniumHQ/docker-selenium/assets/166417911/6c7139ec-9fbf-494c-92ba-bc334b0af0ed">

**Comment by diemol** [comment]
https://github.com/SeleniumHQ/docker-selenium?tab=readme-ov-file#increasing-session-concurrency-per-container

You have `SE_NODE_MAX_SESSIONS` set to `2`. 

Also, `SE_NODE_MAX_INSTANCES` does not exist anymore.

**Reply by MartinVWWA** [reply]
I can ask what to do in a case where it's the second machine on which I would like to run two containers with one session, but they have the same SE_NODE_HOST; otherwise, they don't want to add, but in this case, I don't see them as separate containers in the hub?

![image](https://github.com/SeleniumHQ/docker-selenium/assets/166417911/2f56e004-28c6-463f-ab29-54a737d352ba)

![image](https://github.com/SeleniumHQ/docker-selenium/assets/166417911/78d9c696-4c8b-4ba9-aa62-157029842804)

**Reply by MartinVWWA** [reply]
There's also a strange case when one test ends in a session, and suddenly I see another browser being launched, and after a few seconds, the session gets killed. What could be the reason for this?

**Reply by diemol** [reply]
Why do you need them to have the same `SE_NODE_HOST`?

**Reply by MartinVWWA** [reply]
If I change it for .33 it will be brake on chrome3-1  | 11:37:57.752 INFO [NodeServer$1.lambda$start$1] - Sending registration event... I got this loop

**Reply by diemol** [reply]
Why do you need to use `SE_NODE_HOST`? All containers are in the same docker-compose network.

**Reply by MartinVWWA** [reply]
They are in the same network, but when I didn't add this IP in the logs, I have a loop of this chrome2-1 | 13:01:22.812 INFO [NodeServer$1.lambda$start$1] - Sending registration event..

**Reply by diemol** [reply]
Why don't you use the compose file we offer? https://github.com/SeleniumHQ/docker-selenium/blob/trunk/docker-compose-v3.yml

**Reply by MartinVWWA** [reply]
Sure, I use it when I'm on the same machine and want to set up a node, but here we have a case where it's the second machine that needs to connect to the first one and then have the ability to create nodes, and currently, it only allows creating one using SE_NODE_HOST where I provide the IP of the machine that should create it, and I also provide the IP where the Selenium Hub is located.

**Reply by diemol** [reply]
It is all in the documentation: https://github.com/SeleniumHQ/docker-selenium?tab=readme-ov-file#using-different-machinesvms

**Reply by diemol** [reply]
You just need to make sure TCP connections are allowed between both machines.

**Reply by MartinVWWA** [reply]
Ok I set it up in this way but what when we like to have more that one selenium/node-chrome? 
In this case it work if I have only one  selenium/node-chrome
![image](https://github.com/SeleniumHQ/docker-selenium/assets/166417911/2bc88568-7ff6-4e9a-be8e-1dcbf380e7f5)

**Reply by diemol** [reply]
You can see how the port changes here: https://github.com/SeleniumHQ/docker-selenium?tab=readme-ov-file#node-chrome---machinevm-4

**Reply by MartinVWWA** [reply]
However, I have one more question: where does the issue stem from if during a session, the first test runs and finishes, then another test starts in that same session, but it seems like the first one is closing that session, and I'm not sure why this is happening strangely.

**Reply by diemol** [reply]
I don't understand what you are saying. It is hard to follow when different things are mixed.

**Reply by MartinVWWA** [reply]
Sorry for that

**Reply by MartinVWWA** [reply]
Let's close this thread. I managed to open one node on each machine. Maybe I'll figure out how to handle tests with sessions.

**Comment by MartinVWWA** [comment, marked as answer]
Hi, I found solution for this problem it was the port it should be 5556:5556 not 5555.
![image](https://github.com/SeleniumHQ/docker-selenium/assets/166417911/2299bec0-4d7c-4de4-b739-9e0e0be4e9af)
